src/clients/FtpGateway.cc

```diff
@@ -2639,7 +2645,7 @@ Ftp::Gateway::completeForwarding()
 {
     if (fwd == nullptr || flags.completed_forwarding) {
         debugs(9, 3, "avoid double-complete on FD " <<
-               (ctrl.conn ? ctrl.conn->fd : -1) << ", Data FD " << data.conn->fd <<
+               (ctrl.conn ? ctrl.conn->fd : -1) << ", Data FD " << (data.conn ? data.conn->fd : -1) <<
```

Comment:
This fixes a SIGSEGV occurring when calling Ftp::Gateway::completeForwarding() the second time and trying to dereference data.conn->fd. This code became activated once we started to call completeForwarding() multiple times.

Reply:
Posted as a separate [fix](https://github.com/squid-cache/squid/pull/1923/files).